src/acceleration/impl/WaveformTimeGrids.hpp

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include <Eigen/Core>
+#include <numeric>
+#include <vector>
+
+#include "cplscheme/CouplingData.hpp"
+#include "logging/LogMacros.hpp"
+#include "logging/Logger.hpp"
+#include "utils/assertion.hpp"
+
+namespace precice {
+namespace acceleration {
+namespace impl {
+
+/**
+ * @brief Interface for storing the time grids in the Quasi-Newton and Aitken methods
+ *
+ *
+ * initialize saves the time grids of the waveforms
+ * moveTimeGridToNewWindow Moves the time grid to the new time window. This is done by translating and scaling the QN time grid such that its startpoint and end point is the same as the waveforms
+ *
+ */
+class WaveformTimeGrids {
+public:
+  /// Map from data ID to data values.
+  using DataMap = std::map<int, cplscheme::PtrCouplingData>;
+
+  /**
+   * @brief sets the time grid
+   * @param cplData
+   * @param dataIDs
+   * @param reduced if true then only the last timestep is saved in the timeGrid otherwise all timesteps are saved in the time grid
+   * This is used in the Quasi-Newton method to toggle between the full and reduced variant detailed here
+   */
+  void setTimeGrid(const DataMap &cplData, std::vector<int> dataIDs, bool reduced);
```

Comment:
From a setter, I would expect that I give the time grid as an argument. Could you extend the comment what the function exactly does. Maybe choose a different name?
And shouldn't it be plural as we pass multiple dataIDs?